compute_tools/src/rsyslog.rs

```diff
@@ -0,0 +1,88 @@
+use std::process::Command;
+use std::{fs::OpenOptions, io::Write};
+
+use anyhow::{Context, Result};
+use tracing::info;
+
+fn get_rsyslog_pid() -> Option<String> {
+    let output = Command::new("pgrep")
+        .arg("rsyslogd")
+        .output()
+        .expect("Failed to execute pgrep");
+
+    if !output.stdout.is_empty() {
+        let pid = std::str::from_utf8(&output.stdout)
+            .expect("Invalid UTF-8 in process output")
+            .trim()
+            .to_string();
+        Some(pid)
+    } else {
+        None
+    }
+}
+
+// Start rsyslogd with the specified configuration file
+// If it is already running, do nothing.
+fn start_rsyslog(rsyslog_conf_path: &str) -> Result<()> {
+    let pid = get_rsyslog_pid();
+    if let Some(pid) = pid {
+        info!("rsyslogd is already running with pid: {}", pid);
+        return Ok(());
+    }
+
+    let _ = Command::new("/usr/sbin/rsyslogd")
+        .arg("-f")
+        .arg(rsyslog_conf_path)
+        .arg("-i")
+        .arg("/var/run/rsyslogd/rsyslogd.pid")
+        .output()
+        .context("Failed to start rsyslogd")?;
+
+    // Check that rsyslogd is running
+    if let Some(pid) = get_rsyslog_pid() {
+        info!("rsyslogd started successfully with pid: {}", pid);
+    } else {
+        return Err(anyhow::anyhow!("Failed to start rsyslogd"));
+    }
+
+    Ok(())
+}
+
+pub fn configure_and_start_rsyslog(
+    log_directory: &str,
+    tag: &str,
+    remote_endpoint: &str,
+) -> Result<()> {
+    let config_content = format!(
+        r#"
+# Load imfile module to read log files
+module(load="imfile")
+
+# Input configuration for log files in the specified directory
+input(type="imfile" File="{}/*.log" Tag="{}" Severity="info" Facility="local0")
+global(workDirectory="/var/log")
```

Comment:
I don't see that we configure any specific file name for pgauditlogtofile, should we? Are there any other logs in /var/log, do we stream them all?

I struggle to understand how do we ensure
1. None of audit logs go to the standard stderr
2. Rsyslog understands that it needs to collect only audit logs and nothing else

Reply:
>I don't see that we configure any specific file name for pgauditlogtofile, should we?

>How Rsyslog understands that it needs to collect only audit logs and nothing else

by default pgauditlogtofile writes logs into `pgdata/log/*.log` and we set this via 
`File="{}/*.log" .... log_directory`
where log_directory is pgdata/log

>Are there any other logs in /var/log, do we stream them all?

We don't read anything but audit logs, located in `pgdata/log/*.log`

`workDirectory=/var/log` is the place for rsyslog to store its own work files, i.e. imfile state (last position read from the imfile).

>How do we ensure that None of audit logs go to the standard stderr

`pgauditlogtofile` intercepts them, that's exactly the purpose of this extension.

Reply:
> by default pgauditlogtofile writes logs into pgdata/log/*.log and we set this via

I still do not fully understand what `*.log` mean. Is it some specific file name? Is it one file per-DB? Or per some timeframe? Will these files be rotated? Or they just pile up on the local disk forever?

Reply:
Log files are generated by `pgauditlogtofile`
https://github.com/fmbiete/pgauditlogtofile?tab=readme-ov-file#pgauditlog_filename
Default file names: 'audit-%Y%m%d_%H%M.log'

pgaudit will watch the directory and pick up any files and new rows in the files.

Log files are rotated by the extension, based on `pgaudit.log_rotation_age` setting.
I've set `pgaudit.log_rotation_age=5min`, because default is too conservative.
will tune it after performance testing

Reply:
Sorry for more dumb questions, but who will delete the old files? :) I guess it's not the extension, as it will only switch to a new file. Will rsyslog do that? After brief searching, I cannot find that rsyslog could do that, so people use a separate logrotate for that

Not something to immediately address, but let's track this in the issue